modules/src/agent_deployer/agent_deployer.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+import os
+
+import mlrun.errors
+from mlrun import get_current_project, code_to_function, mlconf
+from mlrun.serving import ModelRunnerStep
+from mlrun.datastore.datastore_profile import DatastoreProfileV3io,DatastoreProfileKafkaStream, DatastoreProfileTDEngine
+
+
+class AgentDeployer:
+    """Class to deploy an agent as a serving function in MLRun."""
+    def __init__(
+            self,
+            agent_name: str,
+            model_class_name: str,
+            function: str,
+            result_path: Optional[str] = None,
+            inputs_path: Optional[str] = None,
+            output_schema: Optional[list[str]] = None,
+            requirements: Optional[list[str]] = None,
+            image: Optional[str] = "mlrun/mlrun",
+            set_model_monitoring: Optional[bool] = False,
```

Comment:
```suggestion
            set_model_monitoring: bool = False,
```

OR 

```suggestion
            set_model_monitoring: Optional[bool] = None,
```

Same for the image param